src/lat/lattice-functions.h

```diff
@@ -28,14 +28,24 @@
 #include <map>
 
 #include "base/kaldi-common.h"
-#include "hmm/posterior.h"
 #include "fstext/fstext-lib.h"
-#include "hmm/transition-model.h"
 #include "lat/kaldi-lattice.h"
 #include "itf/decodable-itf.h"
+#include "itf/transition-information.h"
+#ifndef KALDI_MINIMAL_LATTICE_FUNCTIONS
+#include "lat/lattice-functions-transition-model.h"
+#endif // KALDI_MINIMAL_LATTICE_FUNCTIONS
 
 namespace kaldi {
 
+// Redundant with the typedef in hmm/posterior.h. Redeclaration of a
+// typedef is valid in C++, but not C. This is for the sake of users
+// who compile with the macro KALDI_MINIMAL_LATTICE_FUNCTIONS defined,
+// so that they compile the majority of the lattice functions (all but
+// those declared in lattice-functions-transition-model.h) without depending
+// upon the hmm/ or tree/ libraries.
+typedef std::vector<std::vector<std::pair<int32, BaseFloat> > > Posterior;
```

Comment:
If we're getting rid of `KALDI_MINIMAL_LATTICE_FUNCTIONS`. then probably this comment is redundant? Also, it's obviously C++, so there's hardly any point going on tangent about C typedef rules.